vault/logical_system_test.go

```diff
@@ -315,6 +317,179 @@ func TestSystemBackend_remount_system(t *testing.T) {
 	}
 }
 
+func TestSystemBackend_lease(t *testing.T) {
+	core, b, root := testCoreSystemBackend(t)
+
+	// Create a key with a lease
+	req := logical.TestRequest(t, logical.UpdateOperation, "secret/foo")
+	req.Data["foo"] = "bar"
+	req.ClientToken = root
+	resp, err := core.HandleRequest(req)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if resp != nil {
+		t.Fatalf("bad: %#v", resp)
+	}
+
+	// Read a key with a LeaseID
+	req = logical.TestRequest(t, logical.ReadOperation, "secret/foo")
```

Comment:
It does and I was surprised too and this how all the renew and revoke functions are written.  I think it might be something in the test harness.  Otherwise it isn't straightforward creating a non-token lease without some backing dynamic backend.  Maybe `pki` with `generate_lease` set `true`.